src/h5web/storage-utils.ts

```diff
@@ -0,0 +1,54 @@
+import create, { State, StateCreator, StoreApi, UseStore } from 'zustand';
+
+export interface StorageConfig {
+  key: string;
+  whitelist: string[];
+}
+
+function fromStorage<T>(key: string): T | undefined {
+  try {
+    const value = window.localStorage.getItem(key);
+    return value !== undefined && value !== null
+      ? JSON.parse(value)
+      : undefined;
+  } catch {
+    // Access to local storage may not be authorized or available,
+    // or the data stored may be corrupted.
+    return undefined;
+  }
+}
+
+export function createPersistableState<S extends State>(
+  config: StorageConfig,
+  createState: StateCreator<S>
+): [UseStore<S>, StoreApi<S>] {
+  const { key, whitelist } = config;
+
+  return create<S>((set, get, api) => {
+    // Retrieve persisted state, if any
+    const persistedState = fromStorage<Partial<S>>(key);
+
+    // Invoke provided state creator to retrieve initial state
+    const initialState = createState(
+      // Monkey patch `set` to persist state on every call
+      args => {
+        set(args);
+
+        try {
+          // Retrieve and persist new state
+          const stateStr = JSON.stringify(get(), whitelist); // filter keys to be persisted with whitelist
```

Comment:
Should this line be out of the `try` block to avoid catching errors not linked to the local storage ?

Reply:
totally 🎣